src/context/ShoppingContext.js

```diff
@@ -0,0 +1,23 @@
+import { createContext, useContext, useState } from "react";
+
+const ShoppingContext = createContext();
+
+const ShoppingProvider = ({ children }) => {
+  const [order, setOrder] = useState([]);
+
+  return (
+    <ShoppingContext.Provider
+      value={{
+        order,
+        setOrder,
+      }}
+    >
+      {children}
+    </ShoppingContext.Provider>
+  );
+};
+export const useShopping = () => {
+  return useContext(ShoppingContext);
+};
```

Comment:
please separate this hook in another file and lets add more validation on the code like this :

```suggestion
function useShopping() {
  const context = useContext(ShoppingContext);

  if (!context) {
    throw new Error(
      "useShopping must be used within a ShoppingContextProvider"
    );
  }

  return context;
}
```